caas/kubernetes/clientconfig/base_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package clientconfig_test
+
+import (
+	"github.com/golang/mock/gomock"
+	gc "gopkg.in/check.v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/juju/juju/caas/kubernetes/provider/mocks"
+	"github.com/juju/juju/testing"
+)
+
+type BaseSuite struct {
+	testing.BaseSuite
+
+	namespace string
+
+	ctrl                   *gomock.Controller
+	k8sClient              *mocks.MockInterface
+	mockRbacV1             *mocks.MockRbacV1Interface
+	mockClusterRoles       *mocks.MockClusterRoleInterface
+	mockClusterRoleBinding *mocks.MockClusterRoleBindingInterface
+	mockServiceAccounts    *mocks.MockServiceAccountInterface
+	mockSecrets            *mocks.MockSecretInterface
+}
+
+func (s *BaseSuite) SetUpSuite(c *gc.C) {
+	s.BaseSuite.SetUpSuite(c)
+	s.namespace = "test"
+}
+
+func (s *BaseSuite) setupBroker(c *gc.C) *gomock.Controller {
```

Comment:
+1 for using this pattern. It's a really tidy way to set up base behaviour.

Reply:
@wallyworld  started to use this pattern for k8sBrokerSuite, I like this pattern as well.